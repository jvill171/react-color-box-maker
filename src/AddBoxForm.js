import React, { useState } from "react";

const AddBoxForm = ({addBox}) => {
    const MIN_LENGTH = "35"
    const INITIAL_STATE = {
        "backgroundColor":"#FFFFFF",
        "width": MIN_LENGTH,
        "height": MIN_LENGTH,
    }
    const [formData, setFormData] = useState(INITIAL_STATE);

    const handleChange = (e) =>{
        const {name, value} = e.target
        setFormData(data => ({ ...data, [name]: value }))
    }
    const handleSubmit = (e) =>{
        e.preventDefault()
        // Ensure numeric
        formData.width = parseInt(formData.width)
        formData.height = parseInt(formData.height)
        addBox({...formData})
        setFormData(INITIAL_STATE)
    }
    
    return (
        <form onSubmit={handleSubmit}>
            <label htmlFor="backgroundColor">Color</label>
            <input
                id="backgroundColor"
                name="backgroundColor"
                type="color"
                value={formData["backgroundColor"]}
                onChange={handleChange}
            />
            <br/>

            <label htmlFor="width">Width</label>
            <input
                id="width"
                name="width"
                type="number"
                min={MIN_LENGTH}
                step="5"
                value={formData.width}
                onChange={handleChange}
            />
            <br/>

            <label htmlFor="height">Height</label>
            <input
                id="height"
                name="height"
                type="number"
                min={MIN_LENGTH}
                step="5"
                value={formData.height}
                onChange={handleChange}
            />
            <br/>
            <br/>

            <input type="submit"/>
        </form>
    )
}

export default AddBoxForm;