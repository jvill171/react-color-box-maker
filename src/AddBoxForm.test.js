import { render } from "@testing-library/react"
import AddBoxForm from './AddBoxForm';

// Smoke test
test('renders without crashing', () => {
    render(<AddBoxForm />);
});
  
// Snapshot test
it("matches snapshot", ()=>{
    const { asFragment } = render(<AddBoxForm />);
    expect(asFragment()).toMatchSnapshot()
})