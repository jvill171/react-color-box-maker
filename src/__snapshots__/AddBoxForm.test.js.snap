// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <form
    class="box-form"
  >
    <label
      for="backgroundColor"
    >
      Color
    </label>
    <input
      id="backgroundColor"
      name="backgroundColor"
      type="color"
      value="#FFFFFF"
    />
    <br />
    <label
      for="width"
    >
      Width
    </label>
    <input
      id="width"
      min="2"
      name="width"
      type="number"
      value="2"
    />
    <br />
    <label
      for="height"
    >
      Height
    </label>
    <input
      id="height"
      min="2"
      name="height"
      type="number"
      value="2"
    />
    <br />
    <br />
    <button>
      Submit
    </button>
    u
  </form>
</DocumentFragment>
`;
